hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/NamenodeHeartbeatService.java

```diff
@@ -482,17 +502,35 @@ private void updateHAStatusParameters(NamenodeStatusReport report) {
     }
   }
 
+  /**
+   * Evaluates whether the JMX report should be refreshed by
+   * calling the Namenode, based on the following conditions:
+   * 1. JMX Updates must be enabled.
+   * 2. The last attempt to update JMX occurred before the
+   *    configured interval (if any).
+   */
+  private boolean shouldUpdateJmx() {
+    return updateJmxIntervalMs >= 0
```

Comment:
this.updateJmxIntervalMs and this.lastJmxUpdateAttempt

Reply:
Shouldn't this be:
updateJmxIntervalMs == 0 or Time.monotonicNow() - lastJmxUpdateAttempt > updateJmxIntervalMs
?

Reply:
This wouldn't handle the negative interval properly, which is meant to disable the updates. I added `this.` in a few places